pkg/virtctl/pause/pause.go

```diff
@@ -124,63 +97,45 @@ func (vc *virtCommand) Run(args []string) error {
-			if err != nil {
-				return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", resourceName, err)
-			}
-			printLog(resourceName, vc.command)
+func executePauseCMD(client kubecli.KubevirtClient, namespace, resourceType, resourceName string, dryRunOption []string) error {
+	switch resourceType {
+	case ARG_VM_LONG, ARG_VM_SHORT:
+		vm, err := client.VirtualMachine(namespace).Get(context.Background(), resourceName, v1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Error getting VirtualMachine %s: %v", resourceName, err)
 		}
-	case COMMAND_UNPAUSE:
-		switch resourceType {
-		case ARG_VM_LONG, ARG_VM_SHORT:
-			vm, err := virtClient.VirtualMachine(namespace).Get(context.Background(), resourceName, v1.GetOptions{})
-			if err != nil {
-				return fmt.Errorf("Error getting VirtualMachine %s: %v", resourceName, err)
-			}
-			vmiName := vm.Name
-			err = virtClient.VirtualMachineInstance(namespace).Unpause(context.Background(), vmiName, &kubevirtV1.UnpauseOptions{DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error unpausing VirtualMachineInstance %s: %v", vmiName, err)
-			}
-			printLog(vmiName, vc.command)
-		case ARG_VMI_LONG, ARG_VMI_SHORT:
-			err = virtClient.VirtualMachineInstance(namespace).Unpause(context.Background(), resourceName, &kubevirtV1.UnpauseOptions{DryRun: dryRunOption})
-			if err != nil {
-				return fmt.Errorf("Error unpausing VirtualMachineInstance %s: %v", resourceName, err)
+		vmiName := vm.Name
+		err = client.VirtualMachineInstance(namespace).Pause(context.Background(), vmiName, &kubevirtV1.PauseOptions{DryRun: dryRunOption})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				runningStrategy, err := vm.RunStrategy()
+				if err != nil {
+					return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
+				}
+				if runningStrategy == kubevirtV1.RunStrategyHalted {
+					return fmt.Errorf("Error pausing VirtualMachineInstance %s. VirtualMachine %s is not set to run", vmiName, vm.Name)
+				}
+				return fmt.Errorf("Error pausing VirtualMachineInstance %s, it was not found", vmiName)
 			}
-			printLog(resourceName, vc.command)
+			return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", vmiName, err)
 		}
+		printLog(vmiName)
+
+	case ARG_VMI_LONG, ARG_VMI_SHORT:
+		err := client.VirtualMachineInstance(namespace).Pause(context.Background(), resourceName, &kubevirtV1.PauseOptions{DryRun: dryRunOption})
+		if err != nil {
+			return fmt.Errorf("Error pausing VirtualMachineInstance %s: %v", resourceName, err)
+		}
+		printLog(resourceName)
 	}
+
 	return nil
 }
 
-func printLog(resource string, command string) (int, error) {
-	return fmt.Printf("VMI %s was scheduled to %s\n", resource, command)
+func printLog(resource string) (int, error) {
+	return fmt.Printf("VMI %s was scheduled to %s\n", resource, COMMAND_PAUSE)
```

Comment:
Inline this and move it into `Run`. Also, ideally make it use `cmd.Printf`.

Reply:
done the first half of the sentence